backbone/src/main/java/org/sagebionetworks/researchstack/backbone/ui/step/body/DateQuestionBody.java

```diff
@@ -52,10 +59,34 @@ public DateQuestionBody(Step step, StepResult result) {
             this.dateformatter = FormatHelper.getFormat(DateFormat.MEDIUM, FormatHelper.NONE);
         } else if (format.getStyle() == AnswerFormat.DateAnswerStyle.TimeOfDay) {
             this.dateformatter = FormatHelper.getFormat(FormatHelper.NONE, DateFormat.MEDIUM);
+        } else if (format.getStyle() == AnswerFormat.DateAnswerStyle.MonthYear) {
+            this.dateformatter = new SimpleDateFormat("MM/yyyy");
+            this.dateformatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+            this.calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
+        }
+
+
+        // Working with ResearchStack/Backbone I got a Cast Exception trying to Cast Result as Long been a Double
+        // Fix, check if the Timestamp was set as Double or Long
+        Long savedTimeInMillis = null;
+        if (this.result.getResult() != null) {
+            if (this.result.getResult() instanceof Double) {
+                savedTimeInMillis = ((Double) this.result.getResult()).longValue();
```

Comment:
This seems weird to me that there would be a date or time stored as a double.  Is this workaround masking another problem?